sklearn/svm/base.py

```diff
@@ -169,7 +169,19 @@ def fit(self, X, y, sample_weight=None):
                              "boolean masks (use `indices=True` in CV)."
                              % (sample_weight.shape, X.shape))
 
-        if self.gamma == 'auto':
+        if self.gamma == 'scale':
+            if isinstance(X, sp.spmatrix):
+                X_std = np.sqrt(X.power(2).mean() - (X.mean())**2)
```

Comment:
I've been able to figure out how to do this without the use of `X.power`.

Reply:
I could simply use `X.toarray()**2`, but I don't think that is efficient.

Reply:
Maybe `X.multiply(X)`?

Reply:
Why can there be a ``lil`` matrix here? We called ``check_X_y`` above with ``accept_sparse='csr'``.